packages/@lwc/synthetic-shadow/src/faux-shadow/shadow-root.ts

```diff
@@ -85,6 +87,10 @@ defineProperty(Node.prototype, ShadowRootResolverKey, {
     enumerable: true,
 });
 
+defineProperty(globalThis, '$isNativeShadowRootDefined$', {
```

Comment:
I am wondering about moving those identifier names to `@lwc/shared` to avoid having to keep them in sync between the engine and synthetic shadow would be preferable. Something like:

```js
import { IS_NATIVE_SHADOW_ROOT_DEFINED_FLAG } from '@lwc/shared';

defineProperty(globalThis, IS_NATIVE_SHADOW_ROOT_DEFINED_FLAG, {
  value: isNativeShadowRootDefined
});
```